pkgs/development/interpreters/ruby/parse_version.nix

```diff
@@ -0,0 +1,91 @@
+# Contains the ruby version heuristics
+{ lib }:
+with lib;
+let
+  parseVersion = versionStr:
+    let
+      pair = splitString "-" versionStr;
```

Comment:
Is there a rule against having a '-' in a package name? I couldn't find any and that would break this implementation. Otherwise there are multiple cases of exactly this use of splitString in nixpkgs already.